src/local_neighborhood.py

```diff
@@ -0,0 +1,122 @@
+            node_df = data.request_node_columns(['prize','sources','targets'])
+        elif data.contains_node_columns('prize'):
+            #NODEID is always included in the node table
+            node_df = data.request_node_columns(['prize'])
+        elif data.contains_node_columns(['sources','targets']):
+            #If there aren't prizes but are sources and targets, make prizes based on them
+            node_df = data.request_node_columns(['sources','targets'])
+            node_df.loc[node_df['sources']==True, 'prize'] = 1.0
+            node_df.loc[node_df['targets']==True, 'prize'] = 1.0
+        else:
+            raise ValueError("LocalNeighborhood requires nodes and networks")
+
+        # input node information
+        node_df.to_csv(filename_map["nodes"],sep="\t",index=False,columns=["#Node","Node type"])
+
+        #For now we assume all input networks are undirected until we expand how edge tables work
+        edges_df = data.get_interactome()
+        edges_df.to_csv(filename_map['network'],sep='|',index=False,columns=['vertex1','vertex2'])
+
+
+    # TODO: implement run function
+    @staticmethod
+    def run(network=None, nodes=None, output_file=None, k=None, singularity=False):
+        """
+        Run LocalNeighborhood with Docker
+        @param nodes:  input nodes with sources and targets (required)
+        @param network:  input network file (required)
+        @param output_file: path to the output pathway file (required)
+        @param k: path length (optional)
+        @param singularity: if True, run using the Singularity container instead of the Docker container
+        """
+        if not nodes or not network or not output_file:
+            raise ValueError('Required LocalNeighborhood arguments are missing')
+
+        work_dir = '/spras'
+
+        # Each volume is a tuple (src, dest)
+        volumes = list()
+
+        bind_path, node_file = prepare_volume(nodes, work_dir)
+        print("bind_path: ", bind_path)
+        print("node_file: ", node_file)
+
+        volumes.append(bind_path)
+        print("volumes1: ", volumes)
+
+        bind_path, network_file = prepare_volume(network, work_dir)
+        volumes.append(bind_path)
+        print("volumes2: ", volumes)
+
+        # PathLinker does not provide an argument to set the output directory
+        # Use its --output argument to set the output file prefix to specify an absolute path and prefix
+        out_dir = Path(output_file).parent
+        print("out_dir: ", out_dir)
+        # PathLinker requires that the output directory exist
+        out_dir.mkdir(parents=True, exist_ok=True)
+        bind_path, mapped_out_dir = prepare_volume(str(out_dir), work_dir)
+        volumes.append(bind_path)
+        print("bind_path: ", bind_path)
+        print("mapped_out_dir: ", mapped_out_dir)
+        print("bind_path: ", bind_path)
+        mapped_out_prefix = mapped_out_dir + '/out'  # Use posix path inside the container
```

Comment:
We can simplify the `output_file` setup for local neighborhood algorithm because it takes an output file directly as the command line argument. Therefore it can be treated like the input files `nodes` and passed through `prepare_volume` to get a `mapped_output_file` that is directly provided to the Python command.